File: leetcode/0807-custom-sort-string/solution.go
func customSortString(order string, s string) string {
    var count [26]int
    for _,c := range s {
        count[c-'a']++
    }
    ret := ""
    for _,c := range order {
        for i := 0;i<count[c-'a'];i++ {
            ret += string(c)
        }
        count[c-'a'] = 0
    }
    for i := 0;i<26;i++ {
        for j := 0;j<count[i];j++ {
            ret += string(i+'a')
        }
    }
    return ret
}

File: leetcode/0807-custom-sort-string/solution.rs
impl Solution {
    pub fn custom_sort_string(order: String, s: String) -> String {
        let mut count = vec![0;26];
        for c in s.chars() {
            let c = c as usize - 'a' as usize;
            count[c] += 1;
        }
        let mut ret = String::new();
        for c in order.chars() {
            let k = c as usize - 'a' as usize;
            for i in 0..count[k] {
                ret.push(c);
            }
            count[k] = 0;
        }
        for i in 0..26 {
            let c = (i as u8 + 'a' as u8) as char;
            for j in 0..count[i] {
                ret.push(c);
            }
        }
        ret
    }
}
